lib/experimental/fonts-api/class-wp-fonts-resolver.php

```diff
@@ -214,7 +214,7 @@ private static function get_settings() {
 
 			// Merge the variation settings with the global settings.
 			$settings['typography']['fontFamilies']['theme'] = array_merge(
-				$settings['typography']['fontFamilies']['theme'],
+				$settings['typography']['fontFamilies']['theme'] ?? array(),
```

Comment:
Excuse me, but I don't understand why this change is necessary.
If `$settings['typography']['fontFamilies']['theme']` is `null`, it will be set to an array here (around line 205):
```php
if ( ! isset( $settings['typography']['fontFamilies']['theme'] ) || ! is_array( $settings['typography']['fontFamilies']['theme'] ) ) {
    $settings['typography']['fontFamilies']['theme'] = array();
}
```

Reply:
My best guess is that it's needed because the line you mention depends on us being within the editor: 

https://github.com/WordPress/gutenberg/blob/787f8eade02c93389cf11f7090428f0a5a5b8ba3/lib/experimental/fonts-api/class-wp-fonts-resolver.php#L190

https://github.com/WordPress/gutenberg/blob/787f8eade02c93389cf11f7090428f0a5a5b8ba3/lib/experimental/fonts-api/class-wp-fonts-resolver.php#L200

If I understand that logic correctly, the second time we go through this condition, `$set_theme_structure` could be `false`, hence `$settings['typography']['fontFamilies']['theme']` never getting initialized as an array. Another solution would be to move that initialization out of the conditional on `$set_theme_structure`.

Reply:
> `$settings['typography']['fontFamilies']['theme']` never getting initialized as an array.

Yes, `$set_theme_structure` is `false` on the second iteration of the `foreach` loop. 

However, I don't understand how this affects `$settings['typography']['fontFamilies']['theme']`.
`$settings['typography']['fontFamilies']['theme']` is set to an array during the first iteration of the loop and remains an array in subsequent iterations. `array_merge()` always returns an array value.

Reply:
https://github.com/WordPress/gutenberg/blob/787f8eade02c93389cf11f7090428f0a5a5b8ba3/lib/experimental/fonts-api/class-wp-fonts-resolver.php#L222

After `array_unique` on line 222, `$settings['typography']['fontFamilies']['theme']` is going `null`, and it doesn't get reset in the next iteration. It's probably going `null` because `array_unique` is getting rid of a duplicate `key` that contains the the `theme` array, because the function doesn't check for the value.

Reply:
Hiya! I proposed https://github.com/WordPress/gutenberg/pull/56067 to reset the `$settings['typography']['fontFamilies']['theme']` after the `array_unique` call to avoid it becoming `null` in the next iteration. WDYT?

Reply:
@vcanales 

> After array_unique on line 222, $settings['typography']['fontFamilies']['theme'] is going null, 

I'm not able to reproduce this.

Can you provide the values of `$settings` and `$variations` at which this issue occurs? Please see this code snippet: https://3v4l.org/InOuS#v7.3.0

> After `array_unique` is applied on line 222, `$settings['typography']['fontFamilies']['theme']` becomes null and does not reset in the subsequent iteration.

Can you provide an example of such behavior? Specifically, what exact array should I pass to `array_unique()` in order to make `$settings['typography']['fontFamilies']['theme']` a `null` value?"